### PR TITLE
feat(atomic): new V1 atomic-breadbox component

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-850

Screen Shot 2021-08-20 at 3 55 14 PM
Screen Shot 2021-08-20 at 3 55 01 PM
Screen Shot 2021-08-20 at 3 53 35 PM

### CODE REVIEW

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  @BindStateToController('facetManager')
+  @State()
+  public facetManagerState!: FacetManagerState;
+  @State() public error!: Error;
+  @State() private isCollapsed = true;
+
+  /**
+   * Number of breadcrumbs to display when collapsed.
+   */
+  @Prop() public collapseThreshold = 5;
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:63]: @btaillon + @johnnyCoveo : I guess the same UX principle that was done for multi-text-value should be applied here. Which means that no matter what, we should never be displaying `and one more`.

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    );
+  }
+
+  private get allBreadcrumbs(): Breadcrumb[] {
+    return [
+      ...this.breadcrumbManagerState.facetBreadcrumbs
+        .map(({facetId, field, values}) =>
+          values.map((value) => ({value, facetId, field}))
+        )
+        .flat()
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:161]: you can do `flatMap()` instead on line 158.

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+          text={this.bindings.i18n.t('clear')}
+          class="p-2 btn-pill"
+          ariaLabel={this.bindings.i18n.t('clear-all-filters')}
+          onClick={() => this.breadcrumbManager.deselectAll()}
+        ></Button>
+      </li>
+    );
+  }
+
+  private get allBreadcrumbs(): Breadcrumb[] {
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:155]: There's a lot going on in this function :) 

I'm going to be boring, but just the usual "create N well named function that explains what processing were doing on each type of breadcrumb" would help make this more readable IMO.

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    const sortedBreadcrumbs = this.allBreadcrumbs.sort((a, b) => {
+      const indexA = this.facetManagerState.facetIds.indexOf(a.facetId);
+      const indexB = this.facetManagerState.facetIds.indexOf(b.facetId);
+      return indexA - indexB;
+    });
+    const slicedBreadcrumbs = this.isCollapsed
+      ? sortedBreadcrumbs.slice(0, this.collapseThreshold)
+      : sortedBreadcrumbs;
+
+    return [
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:221]: Similar comment as above, but just old boring code that is not as condensed with small named function would be easier to read.

**Review by olamothe — APPROVED** [review]

**Comment by ThibodeauJF** [comment]
After discussion, I opened https://coveord.atlassian.net/browse/KIT-924, I will make the current option private, merge the PR & work on this other story.

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+  buildFacetManager,
+} from '@coveo/headless';
+import {Button} from '../common/button';
+import CloseIcon from 'coveo-styleguide/resources/icons/svg/close.svg';
+import {getFieldValueCaption} from '../../utils/field-utils';
+
+interface Breadcrumb {
+  facetId: string;
+  label: string;
+  formattedValue: string | string[];
```
Comment (samisayegh) [inline packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:23]: Would it be possible to simplify to only a `string[]` here?

**Review by samisayegh — APPROVED** [review]

**Pull request merged**